osd: Increase priority for inactive PGs backfill

byo opened the pull request:
This change does prioritize backfill of PGs which don't
have min_size active copies. Such PGs would cause IO stalls
for clients and would increase throttlers usage.

This change also fixes few subtlle out-of-bounds bugs.

Signed-off-by: Bartłomiej Święcki <bartlomiej.swiecki@corp.ovh.com>

wido commented:
This PR is very welcome! I would like to propose a backport to Hammer and Jewel when this merges in

liewegas commented:
Do you mind adding a note in PendingReleaseNotes so that we don't forget to include it in the release notes?

byo commented on src/osd/osd_types.h:
```diff
@@ -66,17 +66,24 @@
 
 /// base backfill priority for MBackfillReserve (degraded PG)
-#define OSD_BACKFILL_DEGRADED_PRIORITY_BASE 200u
+#define OSD_BACKFILL_DEGRADED_PRIORITY_BASE 140
```
Hmm, this would be important only if we compare priorities across different OSDs, right? I couldn't find anything in the code that would send this over the network, but maybe I'm not yet that familiar with it ;)

How I was thinking about this: priorities will be consistent within one process and restarting one should also reset all reservations on remote ends so there should be no situation that one OSD would present inconsistent priorities. Only comparing them across different OSDs could go out of control.

byo commented:
Added entry to PendingReleaseNotes

tchaikov commented:
http://pulpito.ceph.com/kchai-2016-12-10_12:27:10-rados-wip-kefu-testing---basic-mira/

byo commented:
@sage: Any chance for backporting this? There's no issue in tracker yet, can make if it's needed.

liewegas commented:
I think so.. please make a ticket

On Thu, 22 Dec 2016, byo wrote:

> 
> @sage: Any chance for backporting this? There's no issue in tracker yet, can make if it's needed.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.[AAabh8ko5kxAxtry6Z-sFaql6LYA7pZUks5rKj2QgaJpZM4LHtFs.gif]
> 
> 
>

byo commented:
Added ticket: http://tracker.ceph.com/issues/18350

I added backport request up till jewel, it can be problematic on hammer since it will not cleanly cherry-pick (some other code changes related to priorities has already been added in jewel).
